CoreMentoringApp.ConsoleRestClient/Program.cs

```diff
@@ -0,0 +1,81 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading.Tasks;
+using CoreMentoringApp.ConsoleRestClient.Models;
+using Newtonsoft.Json;
+
+namespace CoreMentoringApp.ConsoleRestClient
+{
+    class Program
+    {
+        static HttpClient client = new HttpClient();
```

Comment:
Instantiation of HttpClients using `new` keyword is not a good practice in .Net Core world, it leads to:

* hard to test the unit where HttpClient is created with `new` keyword
* socket exhaustion problem
* problem with handling the DNS changes

In .Net Core 2.1 `IHttpClientFactory` was introduced to solve these problems. https://docs.microsoft.com/en-us/dotnet/architecture/microservices/implement-resilient-applications/use-httpclientfactory-to-implement-resilient-http-requests

Since we are learning .net core features, I would ask you to rework this console app client adding to it .Net Core futures as DI and IHttpClient. Here is an example that you can use: https://docs.microsoft.com/en-us/aspnet/core/fundamentals/http-requests?view=aspnetcore-5.0#use-ihttpclientfactory-in-a-console-app

Reply:
Thanks, It was interesting to investigate and apply DI and IHttpClientFactory to console app.